src/scipp/compat/pandas_compat.py

```diff
@@ -3,53 +3,124 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Union
+from typing import TYPE_CHECKING, Callable, Iterable, Literal, Optional, Tuple, Union
 
-from .._scipp.core import DataArray, Dataset, Variable
+from ..core import DataArray, Dataset, Unit, UnitError, array
 from ..typing import VariableLike
+from ..units import default_unit
 
 if TYPE_CHECKING:
     import pandas as pd
 
 
-def from_pandas_series(se: pd.Series) -> DataArray:
+def _index_is_trivial(index: pd.Index, n_rows: int) -> bool:
+    from pandas import RangeIndex
+
+    return (
+        isinstance(index, RangeIndex)
+        and index.start == 0
+        and index.stop == n_rows
+        and index.step == 1
+    )
+
+
+def from_pandas_series(
+    se: pd.Series,
+    *,
+    include_trivial_index: bool = False,
+    header_parser: HeaderParserArg = None,
+) -> DataArray:
     row_index = se.axes[0]
     row_index_name = "row" if row_index.name is None else str(row_index.name)
-    name = "" if se.name is None else str(se.name)
+    name, unit = _parse_header("" if se.name is None else str(se.name), header_parser)
 
+    coords = (
+        {row_index_name: array(dims=[row_index_name], values=row_index)}
+        if include_trivial_index or not _index_is_trivial(row_index, len(se))
```

Comment:
Do you also want to set `unit=None` when the index is trivial?

Reply:
I'd rather not because that would lead to a different unit between, e.g., `RangeIndex(step=1)` and `RangeIndex(step=2)`.